quinn/src/platform/unix.rs

```diff
@@ -223,10 +223,21 @@ fn send(io: &mio::net::UdpSocket, transmits: &[Transmit]) -> io::Result<usize> {
         };
         if n == -1 {
             let e = io::Error::last_os_error();
-            if e.kind() == io::ErrorKind::Interrupted {
-                continue;
+            match e.kind() {
+                io::ErrorKind::Interrupted => {
+                    // Retry the transmission
+                    continue;
+                }
+                io::ErrorKind::PermissionDenied => {
+                    // Transmissions can fail with permission errors for example
+                    // due to iptable rules. Those are not fatal errors, since the
+                    // configuration can be dynamically changed.
+                    // In this case we drop the outgoing packets and let higher
+                    // layers retransmit if required.
+                    return Ok(transmits.len().min(BATCH_SIZE) as _);
```

Comment:
nit: putting this in a temporary and reusing it above would be a bit less error-prone.

Reply:
done